### PR TITLE
remove liquibase-log4j2 module

### PR DESCRIPTION
**Pull request by StevenMassaro**
Remove the liquibase-log4j2 maven module.

The existing module does not work with any recent version of Liquibase, and per previous discussions (see #503) it would be best to remove the code from the log4j2 codebase. Liquibase intends to release and maintain our own implementation of the necessary log4j2 extension.

### CODE REVIEW

**Comment by ppkarwasz** [comment]
@StevenMassaro,

That's great news: `log4j-liquibase` have been lacking contributors for years. Can you post a message on the [`dev@logging.apache.org`](https://logging.apache.org/log4j/2.x/mail-lists.html) mailing list, so that we can coordinate the migration of users to the official Liquibase plugin?

**Comment by StevenMassaro** [comment]
> @StevenMassaro,
> 
> That's great news: `log4j-liquibase` have been lacking contributors for years. Can you post a message on the [`dev@logging.apache.org`](https://logging.apache.org/log4j/2.x/mail-lists.html) mailing list, so that we can coordinate the migration of users to the official Liquibase plugin?

@ppkarwasz , Liquibase has not released their extension for log4j yet. Would that be a prerequisite for removing this code? I was hoping to remove this code since it caused me some confusion on a recent project I was working on, until I realized that it was out of date.

**Comment by jvz** [comment]
We can do it whenever if it's already outdated. The next release even.

**Comment by StevenMassaro** [comment]
> We can do it whenever if it's already outdated. The next release even.

Excellent! Do you need anything else from me?

**Comment by jvz** [comment]
Rebase/merge from current please? :)

**Comment by StevenMassaro** [comment]
> Rebase/merge from current please? :)

Done!

**Comment by StevenMassaro** [comment]
@jvz Is there anything you need from me for this PR before it can be merged?

**Comment by jvz** [comment]
May be helpful to add a changelog entry for this. We just adopted a simpler system for filing them: https://github.com/apache/logging-log4j2/blob/release-2.x/CHANGELOG.adoc

You can file a GitHub Issue to go with this, too, providing additional details. Otherwise, you can simply link to this PR for the changelog entry.

**Comment by StevenMassaro** [comment]
@jvz Just added the changelog entry. I saw that the build failed, though to my untrained eye the failures look unrelated to my changes. Let me know if I'm reading that wrong.

**Comment by jvz** [comment]
Yeah that looks irrelevant, don't worry.

File: src/changelog/.2.x.x/1193_remove_liquibase_log4j2_module.xml [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+-->
+<entry type="changed">
+  <issue id="1193" link="https://github.com/apache/logging-log4j2/pull/1193"/>
+  <author id="StevenMassaro"/>
```
Comment (jvz) [inline src/changelog/.2.x.x/1193_remove_liquibase_log4j2_module.xml:20]: Should use `name` here unless you define your name in the template file for your userid.

Comment (StevenMassaro) [reply src/changelog/.2.x.x/1193_remove_liquibase_log4j2_module.xml]: Resolved, thanks!

**Review by jvz — APPROVED** [review]

**Comment by jvz** [comment]
Thanks for the PR!

**Pull request merged**